### DISCUSSION TITLE
Retrieve Job Object in a trigger listener

### DISCUSSION THREAD

**Discussion by CrazyMoka** [quartznet/quartznet#1699] [Q&A]
Is it possble to get the actually job class that ran by a trigger?
I want to see if I can get some data from that object when the trigger is complete.

**Comment by lahma** [comment]
For example `TriggerFired` gets `IJobExecutionContext context`:

```c#
Type jobType = context.JobDetail.JobType;
IJob instance = context.JobInstance;
JobDataMap dataMap = context.MergedJobDataMap;
```

**Reply by CrazyMoka** [reply]
Thank you but if I get the job instance how can I cast that to my particular job to get access to its methods?
Or what are some use cases for context.JobInstance?

**Reply by lahma** [reply]
You should be able to use the job data map to output state at least.
